index.bs

```diff
@@ -10796,16 +10807,42 @@ operation such as resolution of {{Promise}}s in the {{AudioWorkletGlobalScope}}.
 				[=Computing a block of audio|compute a block of audio=], and
 				[=Making a buffer available for reading|make it available for reading=].
 
-			4. If this {{AudioNode}} is an {{AudioWorkletNode}}, invoke associated
-				{{AudioWorkletProcessor}}'s {{AudioWorkletProcessor/process()}} method
-				to [=Computing a block of audio|compute a block of audio=] with
-				the argument of [=input buffer=], output buffer and
-				[=input AudioParam buffer=].
+			4. If this {{AudioNode}} is an {{AudioWorkletNode}}, execute these
+				substeps:
+
+				1. Let <var>processor</var> be the associated {{AudioWorkletProcessor}}
+					instance of {{AudioWorkletNode}}.
+
+				1. If {{[[callable process]]}} of <var>processor</var> is `true`,
+					execute the following steps:
+
+					1. Let <var>processFunction</var> be the result of
+						<code><a href="http://www.ecma-international.org/ecma-262/6.0/#sec-get-o-p">
+						Get</a>(O=<i><var>processor</var></i>, P="process")</code>.
+
+					1. Set {{[[callable process]]}} to be the return value of
+						<code><a href="http://www.ecma-international.org/ecma-262/6.0/#sec-iscallable">
+						IsCallable</a>(argument=<i><var>processFunction</var></i>)</code>.
+
+					1. If {{[[callable process]]}} is `true`,
+						invoke <var>processFunction</var> to
+						[=Computing a block of audio|compute a block of audio=] with the
+						argument of [=input buffer=], output buffer and
+						[=input AudioParam buffer=].
+
+					1. Else if {{[[callable process]]}} is `false`,
+						<a>Queue a control message</a> to fire `onprocessorerror` event on
```

Comment:
Can we specify the type of error that occurred?

Reply:
It will be `Event` on the main thread, so it is not an actual error event. You can't throw an error on the main thread, because technically an error on the main thread side. What we can do instead is to throw a `TypeError` in AWGS before queueing a message.

Of course, this won't show up (from user land) unless the console in the browser supports the exception on WorkletGlobalScope.

Reply:
I agree, but the `Event` object does not have a valid field to fill with the detailed error message. For the `Event.type`, you'll get `processorerror`.

Also defining a better error reporting for AWP should be a separate piece of work. I've been meaning to do that, but this becomes also useless when you have a proper error handling mechanism in WGS.